lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -2812,7 +2808,16 @@ llvm::Error glow::optimizeFunction(Function *F, const Backend &B,
 
   // Optimize the graph.
   ::glow::optimize(F, cctx);
+  return llvm::Error::success();
+}
 
+// NOTE: When updating this function, please also update the documentation in
+// docs/GraphOptimizationPipeline.md
+llvm::Error glow::optimizeFunction(Function *F, const Backend &B,
+                                   CompilationContext &cctx) {
+  RETURN_IF_ERR(optimizeFunctionBeforeLowering(F, cctx));
+  // Log the start of current log scope.
+  LOG_SCOPE(F->getLogContext(), "glow::optimizeFunction");
```

Comment:
Should this come before the call to `optimizeFunctionBeforeLowering()`?

Reply:
No strong opinion here. If we put it before the call, we'll increase the nesting level of scopes. I can change it if you think it is cleaner.

Reply:
Feels to me like it's more in line with expected uses. @ZchiPitt might have a different feeling. NBD either way 🙂

Reply:
I think it is fine here, as it might reflect the actual context here of the order of calling optimize